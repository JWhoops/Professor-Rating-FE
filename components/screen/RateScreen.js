import * as React from 'react';
import { View, Text, TextInput } from 'react-native';
import { Button, IconButton } from 'react-native-paper';
import { Rating } from 'react-native-ratings';
var Filter = require('bad-words');

class RateScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      termId: "",
      token: "",
      rating: 5,
      comment: '',
    };
  }

  static navigationOptions = ({navigation}) => {
    return({
      title: '',
      headerTitleStyle: {
        color: '#F7F7F7',
      },
      headerStyle: { 
        backgroundColor: '#C5050C',
        shadowOffset: {
          // 设置阴影偏移量
          width: 0,
          height: 4,
        },
        shadowRadius: 4, // 设置阴影模糊半径
        shadowOpacity: 0.13, // 设置阴影的不透明度
        shadowColor: 'rgba(96,96,96,1)', // 设置阴影色
      },
      headerLeft:(()=><IconButton
        icon={'chevron-left'}
        onPress={()=>{navigation.goBack()}}
        color="#F7F7F7"
        size={35}/>)
    })
  };

  onContentSizeChange(event) {
    this.setState({ height: event.nativeEvent.contentSize.height });
  }

  postComment() {
    var filter = new Filter()
    const { token, id } = this.props.navigation.state.params;
    fetch('http://172.220.7.76:8080' + '/v1/comments/', {
      method: 'POST',
      headers: {
        token: token,
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        termId: id,
        rating: this.state.rating,
        content: filter.clean(this.state.comment),
      }),
    })
      .then(response => {
        return response.json();
      })
      .then(result => {
        if (result.success === true) {
          this.props.navigation.state.params.onNavigateBack(id);
          this.props.navigation.goBack();
        } else {
          alert('commentPOST failed, result.sucess: ' + result.success);
        }
      })
      .catch(err => {
        alert(err);
      });
  }

  render() {
    const { course_name } = this.props.navigation.state.params;
    return (
      <View
        style={{
          height: '100%',
          alignItems: 'center',
        }}>
        <View
          style={{
            shadowOffset: {
              width: 0,
              height: 4,
            },
            shadowRadius: 10,
            shadowOpacity: 0.13,
            elevation: 0.2,
            borderRadius: 0,
            shadowColor: 'rgba(96,96,96,1)',
            width: '95%',
            height: 'auto',
            padding: 16,
            marginTop: 15,
            marginHorizontal: 8,
            backgroundColor: '#fff',
          }}>
          <View style={{ flexDirection: 'row', marginHorizontal: 10 }}>
            <Text style={{ fontWeight: 'bold', color: '#666', top: -1.5, fontSize: 17 }}>{course_name}</Text>
            <View style={{ position: 'absolute', right: -5 }}>
              <Rating
                ratingCount={5}
                startingValue={5}
                imageSize={18}
                readyOnly={false}
                onFinishRating={rate => this.setState({ rating: rate })}
              />
            </View>
          </View>
          <View style={{
            backgroundColor: '#fafafa',
            padding: 10,
            marginTop: 15,
            marginBottom: 10,
            width: '100%',
          }}>
            <TextInput
              placeholder="Type here"
              multiline={true}
              onChangeText={text => this.setState({ comment: text })}
              onContentSizeChange={this.onContentSizeChange.bind(this)}
            />
          </View>
        </View>
        <Button
          style={{
            width: '95%',
            marginVertical: 13,
            shadowOffset: {
              width: 0,
              height: 4,
            },
            shadowRadius: 4,
            shadowOpacity: 0.13,
            elevation: 2,
            borderWidth: 0,
            borderRadius: 0,
            padding: 2,
            backgroundColor: '#c5050c',
          }}
          onPress={() => this.postComment()}
          mode="contained">
          SUBMIT
      </Button>
      </View>
    );
  }
}

export default RateScreen;