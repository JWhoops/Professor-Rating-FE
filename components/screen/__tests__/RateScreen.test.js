import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import RateScreen from '../RateScreen';


test('RateScreen renders all inputs as expected', () => {
    const createTestProps = (props) => ({
        navigation: {
          state: { params: {
            course_name: 'COMP SCI 506', 
            token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1ZTdmMWE1Y2RkNjdlMTE5ZTM2MDU1NGQiLCJpYXQiOjE1ODY0NzA1MTd9.wymnro8wu4OD9K5FLZJlCX6zXMJxUI5qg4CJ3kqKJFY',
            id: 'testing id'
          } },
          navigate: jest.fn(),
        },
        ...props
      });
    
      let props = createTestProps();
      const { toJSON } = render(<RateScreen {...props}/>);

  expect(toJSON()).toMatchSnapshot();
});
