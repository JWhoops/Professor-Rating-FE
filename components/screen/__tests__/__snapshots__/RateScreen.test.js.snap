// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateScreen renders all inputs as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 4,
        "height": "auto",
        "marginHorizontal": 8,
        "marginTop": 10,
        "padding": 16,
        "shadowColor": "rgba(96,96,96,1)",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.13,
        "shadowRadius": 4,
        "width": "95%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#666",
            "fontWeight": "bold",
          }
        }
      >
        COMP SCI 506
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": -5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 4,
          "marginBottom": 10,
          "marginTop": 15,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        placeholder="Type here"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginVertical": 13,
        "minWidth": 64,
        "padding": 2,
        "shadowColor": "rgba(96,96,96,1)",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.13,
        "shadowRadius": 4,
        "width": "95%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          SUBMIT
        </Text>
      </View>
    </View>
  </View>
</View>
`;
